src/library-authoring/component-info/ManageCollections.tsx

```diff
@@ -0,0 +1,206 @@
+} from '@openedx/paragon';
+import { Folder } from '@openedx/paragon/icons';
+
+import {
+  ContentHit,
+  SearchContextProvider,
+  SearchKeywordsField,
+  SearchSortWidget,
+  useSearchContext,
+} from '../../search-manager';
+import messages from './messages';
+import { useUpdateComponentCollections } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+interface ManageCollectionsProps {
+  contentHit: ContentHit;
+}
+
+interface CollectionsDrawerProps extends ManageCollectionsProps {
+  onClose: () => void;
+}
+
+const CollectionsSelectableBox = ({ contentHit, onClose }: CollectionsDrawerProps) => {
+  const type = 'checkbox';
+  const intl = useIntl();
+  const { collectionHits } = useSearchContext();
+  const { showToast } = useContext(ToastContext);
+  const [selectedCollections, {
+    add,
+    remove,
+    set,
+    clear,
+  }] = useCheckboxSetValues(contentHit.collections?.key || []);
+
+  useEffect(() => {
+    set(contentHit.collections?.key || []);
+
+    return () => {
+      clear();
+    };
+  }, [contentHit]);
+
+  const updateCollectionsMutation = useUpdateComponentCollections(contentHit.contextKey, contentHit.usageKey);
+
+  const handleConfirmation = () => {
+    updateCollectionsMutation.mutateAsync(selectedCollections).then(() => {
+      showToast(intl.formatMessage(messages.manageCollectionsToComponentSuccess));
+      onClose();
+    }).catch(() => {
+      showToast(intl.formatMessage(messages.manageCollectionsToComponentFailed));
+    });
+  };
+
+  const handleChange = (e) => (e.target.checked ? add(e.target.value) : remove(e.target.value));
+
+  return (
+    <Stack gap={4}>
+      <Scrollable className="mt-3 p-1 border-bottom border-gray-100" style={{ height: '25vh' }}>
+        <SelectableBox.Set
+          value={selectedCollections}
+          type={type}
+          onChange={handleChange}
+          name="selectedCollections"
+          columns={1}
+          ariaLabelledby={intl.formatMessage(messages.manageCollectionsSelectionLabel)}
+        >
+          {collectionHits.map((collectionHit) => (
+            <SelectableBox
+              className="d-inline-flex align-items-center shadow-none border border-gray-100"
+              value={collectionHit.blockId}
+              key={collectionHit.blockId}
+              inputHidden={false}
+              type={type}
+              aria-label={collectionHit.displayName}
+            >
+              <Stack className="ml-2" direction="horizontal" gap={2}>
+                <Icon src={Folder} />
+                <span>{collectionHit.displayName}</span>
+              </Stack>
+            </SelectableBox>
+          ))}
+        </SelectableBox.Set>
+      </Scrollable>
+      <Stack direction="horizontal" gap={2}>
+        <Button
+          onClick={onClose}
+          variant="outline-primary"
+        >
+          {intl.formatMessage(messages.manageCollectionsToComponentCancelBtn)}
+        </Button>
```

Comment:
Nit: The style here is a bit different than the one used in the Manage Tags feature

Reply:
The designs given in the ticket show button with border but for consistency, modified it to match Manage tags `Cancel` button.

![image](https://github.com/user-attachments/assets/3124a95d-3b70-4eb9-b086-b2f70643ddb1)